File: inventory-system/inventory.go
package main

import (
    "fmt"
	"strconv"
)

// Inventaris awal kosong
type item map[string]string
var inventaris []item

// Fungsi untuk menambahkan produk ke inventaris
func tambah_produk(nama string, harga string, jumlah string){
	new_item := make(item)
	new_item["nama"] = nama
	new_item["harga"] = harga
	new_item["jumlah"] = jumlah
    inventaris = append(inventaris, new_item)
}

// Fungsi untuk memperbarui jumlah produk
func perbarui_jumlah(nama_produk string, jumlah_baru string){
	for _, item := range inventaris { 
		if nama_produk == item["nama"] {
			item["jumlah"] = jumlah_baru 
		}
	}
}

// Fungsi untuk menampilkan inventaris saat ini
func tampilkan_inventaris(){
    fmt.Println("Inventaris Saat Ini:")
    fmt.Println("No || Nama || Harga || Jumlah")
    for i, item := range inventaris { 
		fmt.Println(i + 1, item["nama"], item["harga"], item["jumlah"])
	}
}

// Fungsi untuk menghitung total nilai inventaris
func hitung_total_nilai(){
    var total_nilai int64 = 0
	tampilkan_inventaris()
	for _, item := range inventaris {
		harga, _ := strconv.ParseInt(item["harga"], 0, 64)
		jumlah, _ := strconv.ParseInt(item["jumlah"], 0, 64)
		total_nilai += harga * jumlah
	}
    fmt.Println("Total nilai inventaris: ", total_nilai)
}

func main() {
	tambah_produk("Sandal", "50000", "5")
	tambah_produk("Sepatu", "150000", "10")
	tambah_produk("Tas", "100000", "5")
	tampilkan_inventaris()

	perbarui_jumlah("Tas", "15")
	tampilkan_inventaris()

	hitung_total_nilai()
}

File: inventory-system/inventory.py
# Inventaris awal kosong
inventaris = []

# Fungsi untuk menambahkan produk ke inventaris
def tambah_produk(nama, harga, jumlah):
    inventaris.append({'nama': nama, 'harga': harga, 'jumlah': jumlah})

# Fungsi untuk memperbarui jumlah produk
def perbarui_jumlah(nama_produk, jumlah_baru):
    for i in inventaris:
        if i['nama'] == nama_produk:
            i['jumlah'] = jumlah_baru

# Fungsi untuk menampilkan inventaris saat ini
def tampilkan_inventaris():
    print("Inventaris Saat Ini:")
    for i in inventaris:
        print(i)

# Fungsi untuk menghitung total nilai inventaris
def hitung_total_nilai():
    total_nilai = 0
    # proses untuk menghitung total nilai
    for i in inventaris:
        total_nilai += i['harga'] * i['jumlah']
    print("Total nilai inventaris: ", total_nilai)

tambah_produk('sepatu', 50000, 2)
tambah_produk('sandal', 15000, 5)
tambah_produk('tas', 25000, 15)
perbarui_jumlah('sepatu', 10)
tampilkan_inventaris()
hitung_total_nilai()